main.tf

```
@@ -0,0 +1,122 @@
+terraform {
+  required_version = ">= 0.11.2"
+
+  backend "s3" {}
+}
+
+provider "aws" {
+  assume_role {
+    role_arn = "${var.aws_assume_role_arn}"
+  }
+}
+
+variable "aws_assume_role_arn" {
+  type = "string"
+}
+
+variable "name" {
+  type        = "string"
+  description = "Name  (e.g. `codefresh`)"
+  default     = "codefresh"
+}
+
+variable "namespace" {
+  type        = "string"
+  description = "Namespace (e.g. `eg` or `cp`)"
+}
+
+variable "stage" {
+  type        = "string"
+  description = "Stage (e.g. `prod`, `dev`, `staging`)"
+}
+
+variable "zone_name" {
+  type        = "string"
+  description = "DNS zone name"
+}
+
+variable "kms_key_id" {
+  type        = "string"
+  default     = ""
+  description = "KMS key id used to encrypt SSM parameters"
+}
+
+variable "chamber_format" {
+  default     = "/%s/%s"
+  description = "Format to store parameters in SSM, for consumption with chamber"
+}
+
+variable "chamber_service" {
+  default     = ""
+  description = "`chamber` service name. See [chamber usage](https://github.com/segmentio/chamber#usage) for more details"
+}
+
+variable "overwrite_ssm_parameter" {
+  type        = "string"
+  default     = "true"
+  description = "Whether to overwrite an existing SSM parameter"
+}
+
+variable "delimiter" {
+  type        = "string"
+  default     = "-"
+  description = "Delimiter to be used between `namespace`, `stage`, `name` and `attributes`"
+}
+
+variable "attributes" {
+  type        = "list"
+  default     = []
+  description = "Additional attributes (e.g. `1`)"
+}
+
+variable "tags" {
+  type        = "map"
+  default     = {}
+  description = "Additional tags (e.g. map(`Cluster`,`us-east-1.cloudposse.co`)"
+}
+
+variable "node_security_groups" {
+  type        = "list"
+  default     = []
+  description = "List of Kops node security groups to be allowed to connect to the CodeFresh backing infra"
```

```suggestion
  description = "List of security groups to be allowed to connect to the CodeFresh backing infra"
```

Is this module only for kops?

---

Nope, I have renamed the `node_security_groups` variable to be more generic and removed kops related wording.